exportlegends.lua

```diff
@@ -47,6 +77,31 @@ local MAPS = {
     "Diplomacy",
 }
 
+-- Get that date of the world as a string
+-- Format: "YYYYY-MM-DD"
+function get_world_date_str()
+    local month = dfhack.world.ReadCurrentMonth() + 1 --days and months are 1-indexed
+    local day = dfhack.world.ReadCurrentDay()
+    local year_str = string.format('%0'..math.max(5, string.len(''..df.global.cur_year))..'d', df.global.cur_year)
```

Comment:
One last bit of advice, in case it's useful in the future: the `5` in `%05d` is a minimum length, not a minimum and maximum, so you can avoid calculating the length of cur_year here:
```lua
> string.format('%05d', 1)
00001
> string.format('%05d', 125)
00125
> string.format('%05d', 123456)
123456
> string.format('%05d', 1234567890)
1234567890
```

Reply:
Yeah good point, this was just moved from the original code and did not change it. I though there was some reason behind it and did not look much more into it.
Original location: https://github.com/DFHack/scripts/blob/0.47.04-r1/exportlegends.lua#L118-L121
Looks much better in your changes below.

Reply:
Oops, evidently I didn't look at the full diff. Makes sense!